genai-perf/tests/test_cli.py

```diff
@@ -772,9 +772,7 @@ def test_goodput_args_warning(self, monkeypatch, args, expected_error):
     def test_inferred_prompt_source(
         self, monkeypatch, mocker, args, expected_prompt_source
     ):
-        _ = mocker.patch("builtins.open", mocker.mock_open(read_data="data"))
-        _ = mocker.patch("os.path.isfile", return_value=True)
-        _ = mocker.patch("os.path.isdir", return_value=True)
+        _ = mocker.patch.object(Path, "is_file", return_value=True)
```

Comment:
1. why is this an assignment
2. why did the other mocks in the before section get removed?

Reply:
I'll remove the assignment. The other mocks were not necessary. I think what had happened was that we have a `[path.is_file() or path.is_dir](https://github.com/triton-inference-server/perf_analyzer/blob/8c0c09b79e613c809c5d6aded7ea14544e449de8/genai-perf/genai_perf/parser.py#L380)` check... and the way Python works, is_dir was an address and always evaluated to true. The checker caught that, so I had to make it `path.is_dir()` to properly be a function. 

Fixing the static error broke the test... and in fixing it, I realized that I had to mock Path directly (because of how we use it) and the other mocked functions no longer get called.

Reply:
Removed the assignment: https://github.com/triton-inference-server/perf_analyzer/pull/143/commits/9ffff7d5ca301cfe3a14f7cfb3b0a1219c39ef8c